test/e2e/fishapp/load/load_stack_test.go

```diff
@@ -0,0 +1,279 @@
+		var out bytes.Buffer
+		var stderr bytes.Buffer
+		portForwardCmd.Stdout = &out
+		portForwardCmd.Stderr = &stderr
+
+		fmt.Printf("Running port-forwarding command -: %s\n", portForwardCmd.String())
+		portForwardErr := portForwardCmd.Start()
+		if portForwardErr != nil {
+			fmt.Printf("Fortio port-forwarding failed with error -: %v -: %s\n", portForwardErr, stderr.String())
+		} else {
+			fmt.Println(out.String())
+		}
+		Expect(portForwardErr).NotTo(HaveOccurred())
+	})
+
+	By("Adding all VirtualServices as backends of Fortio", func() {
+		var vnBackends []appmesh.Backend
+		for _, vs := range stacksPendingCleanUp[0].createdServiceVSs {
+			fmt.Printf("Granting backend access for Fortio to service -: %s\n", vs.Name)
+			vnBackends = append(vnBackends, appmesh.Backend{
+				VirtualService: appmesh.VirtualServiceBackend{
+					VirtualServiceRef: &appmesh.VirtualServiceReference{
+						Namespace: aws.String(vs.Namespace),
+						Name:      vs.Name,
+					},
+				},
+			})
+		}
+		vnNew := fortio_vn.DeepCopy()
+		fmt.Printf("Number of backends for Fortio = %d\n", len(vnBackends))
+		vnNew.Spec.Backends = vnBackends
+		//if s.IsmTLSEnabled {
+		//	vnNew.Spec.BackendDefaults.ClientPolicy.TLS.Validation.SubjectAlternativeNames.Match.Exact = backendSANs
+		//}
+
+		err := f.K8sClient.Patch(ctx, vnNew, client.MergeFrom(fortio_vn))
+		Expect(err).NotTo(HaveOccurred())
+		stacksPendingCleanUp[0].createdNodeVNs["fortio"] = vnNew
+	})
+
+	return portForwardCmd
+}
+
+var _ = Describe("Running Load Test Driver", func() {
+	var ctx context.Context
+	var f *framework.Framework
+
+	Context("normal test dimensions", func() {
+		var stackPrototype DynamicStackNew
+		var stacksPendingCleanUp []*DynamicStackNew
+
+		BeforeEach(func() {
+			ctx = context.Background()
+			f = framework.New(framework.GlobalOptions)
+
+			// Ginkgo flag parsing happens in BeforeEach or It blocks only
+			configPath = filepath.Join(basePath, "config.json")
+			loadDriverPath = filepath.Join(basePath, "scripts", "load_driver.py")
+			fmt.Println("Reading config from -: ", configPath)
+			config := readDriverConfig(configPath)
+			stackPrototype, stacksPendingCleanUp = createResourcesStack(config)
+		})
+
+		sdType := manifest.DNSServiceDiscovery
+
+		It(fmt.Sprintf("should behaves correctly with service discovery type %v", sdType), func() {
+
+			checkConnectivity := false
+			stacksPendingCleanUp = spinUpResources(stackPrototype, stacksPendingCleanUp, sdType, checkConnectivity, ctx, f)
+			defer spinDownResources(stacksPendingCleanUp, ctx, f)
+
+			portForwardCmd := deployFortio(ctx, f, stacksPendingCleanUp)
+			defer func() {
+				fmt.Println("killing Fortio port-forward process")
+				if portForwardKillErr := portForwardCmd.Process.Kill(); portForwardKillErr != nil {
+					fmt.Println("failed to kill process. Error -: ", portForwardKillErr, "Continuing...")
+				}
+			}()
+
+			time.Sleep(2 * time.Minute)
```

Comment:
Even though I'm waiting for Fortio components to become active and port-forwarding to run fine, if I comment out this sleep, the POST request inside the Python Load driver script fails with Connection Refused. Works fine after a brief delay/sleep

Reply:
We currently have no way to coordinate Envoy start and Application start, so this is (sadly) somewhat expected...